### ISSUE TITLE
Service-discovery deletes the pool members for each sd modification

### ISSUE THREAD

**Issue by jrooty** [F5Networks/f5-appsvcs-extension#788]
This request is related to the service-discovery behavior.

On service-discovery configuration, changing  `uri` or `servicePort`, the current behavior is to delete all pool members (retrieved previously) and wait until the next discovery (based on `updateInterval`) to get new pool members.

Based on that, changing either `uri` or `servicePort` will disturb the traffic.

What we see also, based on new feature #759, if we change the content of `adminState` (could be `enabe`, `disable` or `offline`), we observe the same behavior : all pool members are removed until the next pooling.
That behavior makes useless #759 as we lost all pool members...

I understand that if we modify `uri` or `servicePort`, we should expect new members as we will target new service-discovery service.
However, if we change `adminState`, `priorityGroup` or other, it shouldn't impact the service (why remove pool members?)

We need a behavior change at least for `adminState`, `priorityGroup` or other : do not delete the current pool members. It does make no sence to delete pool members here.  

If technically, it is possible to apply this change for `uri` or `servicePort`, it can be helpfull as well.

Here an extract of the AS3 declaration : 
```
members": [
                    {
                        "servicePort": 30988,
                        "addressDiscovery": "consul",
                        "autoPopulate": true,
                        "uri": https://first.sd.com/,
                        "updateInterval": 30,
                        "rejectUnauthorized": false,
                        "shareNodes": true,
                        "adminState": "disable",
                        "priorityGroup": 10
                    },
                    {
                        "servicePort": 30988,
                        "addressDiscovery": "consul",
                        "autoPopulate": true,
                        "uri": https://second.sd.com/,
                        "updateInterval": 30,
                        "rejectUnauthorized": false,
                        "shareNodes": true,
                        "adminState": "enable",
                        "priorityGroup": 5
                    }
                ]
```

Thank you.

Regards,

**Comment by mdditt2000** [comment]
thanks for the notification via email. Opened AUTOTOOL-4127 and added to the next sprint. As the team to recreate as well.

**Comment by sunitharonan** [comment]
This has been fixed in AS3 50.0 which is planned to release by March end. Please reach out to us if an early build is required.
